iped-parsers/iped-parsers-impl/src/main/java/iped/parsers/shareaza/ShareazaDownloadParser.java

```diff
@@ -0,0 +1,940 @@
+
+            int ed2kListSerialSize = getEd2kListSerialSize(ed2kListSize);
+            addLine(xhtml, "     EDONKEY [HASH DATABASE SIZE]:       " + ed2kListSerialSize);
+
+            padding(buffer, ed2kListSerialSize);
+
+            if (ed2kListSize > 0) {
+                int hashsetBlock = read4Bytes(buffer);
+                int hashsetSuccess = read4Bytes(buffer);
+
+                addLine(xhtml, "     EDONKEY [HASHSET BLOCK LENGTH]:     " + hashsetBlock);
+                addLine(xhtml, "     EDONKEY [HASHSET SUCCESS]:          " + hashsetSuccess);
+
+                padding(buffer, hashsetBlock);
+            }
+
+            int numPreviews = read2Bytes(buffer);
+            addLine(xhtml, "NUMBER OF PREVIEWS:      " + numPreviews);
+
+            for (int i = 0; i < numPreviews; i++) {
+                String preview = readString(buffer);
+                addLine(xhtml, "    PREVIEW " + (i + 1) + ":           " + preview);
+
+            }
+
+            if (version >= 32) {
+                int numReviews = read2Bytes(buffer);
+                addLine(xhtml, "NUMBER OF REVIEWS:       " + numReviews);
+
+                if (numReviews > 0) {
+                    for (int i = 0; i < numReviews; i++) {
+                        addLine(xhtml, "     REVIEW " + (i + 1) + ":");
+
+                        long userIP = read8Bytes(buffer);
+                        int userPicture = read4Bytes(buffer);
+                        String userName = readString(buffer);
+                        int fileRating = read4Bytes(buffer);
+                        String fileComments = readString(buffer);
+
+                        addLine(xhtml, "     	USER IP:       " + userIP);
+                        addLine(xhtml, "     	USER PICTURE:  " + userPicture);
+                        addLine(xhtml, "     	USER NAME:     " + userName);
+                        addLine(xhtml, "     	FILE RATING:   " + fileRating);
+                        addLine(xhtml, "     	FILE COMMENTS: " + fileComments);
+                    }
+                }
+            }
+
+            int expanded = read4Bytes(buffer);
+            int paused = read4Bytes(buffer);
+            int boosted = read4Bytes(buffer);
+            int shared = read4Bytes(buffer);
+            String serialID = readHashString(buffer, 4);
+
+            addLine(xhtml, "EXPANDED:                " + getBoolStr(expanded));
+            addLine(xhtml, "PAUSED:                  " + getBoolStr(paused));
+            addLine(xhtml, "BOOSTED:                 " + getBoolStr(boosted));
+            addLine(xhtml, "SHARED:                  " + getBoolStr(shared));
+            addLine(xhtml, "SERIAL ID:               " + serialID);
+
+        } catch (BufferUnderflowException ex) {
+            addLine(xhtml, INCOMPLETE_FILE_EX_MESSAGE + " Evidence=" + evidenceName);
+            LOGGER.error(INCOMPLETE_FILE_EX_MESSAGE + " Evidence=" + evidenceName);
```

Comment:
I my opinion, if any parse could be done, with at least the name of the download, no log should be output to log file, as it was not an error of the parser, but a incomplete or inconsistent file. The "incomplete" warning could be inside the generated output only.

Reply:
IMHO I think it's fine, we log warnings for corrupted files. But this logging should be warn, error goes to console and would confuse users. I would also log the full file path, not just its name, path can be get from ItemInfo.class from ParseContext.

Reply:
I believe the full path of the file is more appropriate, I will include this change and change the log to warn.